source/fe/fe_system.cc

```diff
@@ -1587,6 +1587,61 @@ void FESystem<dim,spacedim>::initialize (const std::vector<const FiniteElement<d
         }
     });
 
+  // Initialize generalized support points and an (internal) index table
+  init_tasks += Threads::new_task ([&]()
+  {
+    // if one of the base elements has no generalized support points, then
+    // it makes no sense to define generalized support points for the
+    // composed element, so return an empty array to demonstrate that fact.
+    // Note that we ignore FE_Nothing in this logic.
+    for (unsigned int base_el=0; base_el < this->n_base_elements(); ++base_el)
+      if (!base_element(base_el).has_generalized_support_points() &&
+          base_element(base_el).dofs_per_cell != 0)
+        {
+          this->generalized_support_points.resize(0);
+          return;
+        }
+
+    // Iterate over all base elements, extract a representative set of
+    // unique generalized support points and store the information how
+    // generalized support points of base elements are mapped to this list
+    // of representatives.
+
+    generalized_support_points_index_table.resize(this->n_base_elements());
+
+    for (unsigned int base_el = 0; base_el < this->n_base_elements(); ++base_el)
+      {
+        const auto &element = base_element(base_el);
+        const auto &support_points = element.get_generalized_support_points();
+
+        auto &index_table = generalized_support_points_index_table[base_el];
```

Comment:
I find aliases terribly confusing. I think the code may be easier to read if you just worked with the original table because then you get to see how it is filled, without having to unravel which array it is that's currently being filled.

Reply:
I would argue the opposite - but that's nothing I want to discuss on github at 1pm in the night.
I have refactored the code.